Fix #894: links to ietf-wg-acme draft

tdelmas opened the pull request:
TODO:
Update `lastmod` when needed
Check all links

tdelmas commented:
- ACME: https://github.com/ietf-wg-acme/acme => https://tools.ietf.org/html/rfc8555
- ACME Challenges: https://github.com/ietf-wg-acme/acme/blob/master/draft-ietf-acme-acme.md#responding-to-challenges => https://tools.ietf.org/html/rfc8555#section-7.5.1
- DNS Challenge: https://ietf-wg-acme.github.io/acme/#rfc.section.8.5 => https://tools.ietf.org/html/rfc8555#section-8.4
- HTTP Challenge: (https://ietf-wg-acme.github.io/acme/#rfc.section.8.3 => https://tools.ietf.org/html/rfc8555#section-8.3
- TLS Challenge: https://ietf-wg-acme.github.io/acme/#rfc.section.8.4 => https://tools.ietf.org/html/draft-ietf-acme-tls-alpn-07#section-3 

I have doubt about this one, in revoking.md, because I suspect it talked about TLS-SNI-01, but given the context, talking about TLS-ALPN-01 is the good thing to do. but maybe the link of the text needs to be updated as well. It currently says:
> Let's Encrypt uses the same methods it uses to validate control for issuance: you can [...] or offer a [special TLS certificate]

(link on `[special TLS certificate]`) : https://letsencrypt.org/docs/revoking/#using-a-different-authorized-account

can I have a confirmation @jsha ?

jsha commented:
> I have doubt about this one, in revoking.md, because I suspect it talked about TLS-SNI-01, but given the context, talking about TLS-ALPN-01 is the good thing to do. but maybe the link of the text needs to be updated as well. It currently says:

Yes, agreed that this was talking about TLS-SNI. I think the easiest thing to do is just remove the clause talking about a "special TLS certificate." TLS-ALPN is much more of a niche use case than TLS-SNI was, and I think the folks who already authenticate with TLS-ALPN will understand they can use it for revocation too. So updating this to just talk about DNS and HTTP makes sense.

tdelmas commented:
Thanks, I'll update the PR accordingly.

tdelmas commented:
I've removed the clause talking about a "special TLS certificate." in the English version of `revoking.md` and updated `lastmod`. For it's translation, I've kept the updated link, I felt it was better than an invalid link until the translation is updated.

bdaehlie commented:
Once conflicts are resolved you can go ahead and merge this. Thanks!
